runtime/signals_nat.c

```diff
@@ -239,6 +239,12 @@ DECLARE_SIGNAL_HANDLER(segv_handler)
     Caml_state->young_ptr = (value *) CONTEXT_YOUNG_PTR;
 #endif
     caml_raise_stack_overflow();
+#endif
+  } else if (Caml_state->checking_pointer_pc) {
+#ifdef CONTEXT_PC
+    CONTEXT_PC = (context_reg)Caml_state->checking_pointer_pc;
```

Comment:
As the Travis run shows, this line causes an error on i386 Linux, because `context_reg` is not defined.  Moreover, on this particular platform, I'm not sure that assigning `CONTEXT_PC` actually changes the return address for the signal handler.  The reason is that signal handling on i386 Linux is different from that on other Linux platforms and uses an archaic way to obtain the signal context.  

I could try to modernize i386 Linux signal handling.  On the other hand, if this instrumented runtime is intended to be used only for x86-64, it may not be needed.

Reply:
`rr` doesn't work on i386. In fact, it only works on modern Intel processors. So modernizing i386 Linux signal handling may not directly help this PR. 

To suppress the error, I can define `HAS_CONTEXT_REG` on supported architectures and wrap this code with this macro. The end result would be that on i386, this PR would segfault. This may be acceptable only to get the CI to go green.

Reply:
If this is intended to run on x86-64 only, no change is required.  CI can remain red if we understand why!  Still, I might try to clean up i386 signal handling, because I'm not proud of the current code.

Reply:
For the record: #9543 brings the i386 Linux port in line with the other Linux ports concerning signal handling.

Reply:
> For the record: #9543 brings the i386 Linux port in line with the other Linux ports concerning signal handling.

Excellent!